### PR TITLE
GUI: Warn when enabling autosave on options

### PR DESCRIPTION
**Pull request by orgads**
...if non-autosave games are stored in autosave slot

### CODE REVIEW

**Comment by bluegr** [comment]
Looks to be a very straightforward implementation. Nice work! +1 from me

**Comment by bluegr** [comment]
Typo on the commit message: enaling -> enabling

File: gui/options.cpp [hunk 1]
```diff
@@ -2476,6 +2477,83 @@ void GlobalOptionsDialog::addAccessibilityControls(GuiObject *boss, const Common
...
+
+bool GlobalOptionsDialog::updateAutosavePeriod(int newValue) {
+	const int oldAutosavePeriod = ConfMan.getInt("autosave_period");
+	if (oldAutosavePeriod != 0 || newValue <= 0)
+		return true;
+	typedef Common::Array<ExistingSave> ExistingSaveList;
+	ExistingSaveList saveList;
+	using Common::ConfigManager;
+	const ConfigManager::DomainMap &domains = ConfMan.getGameDomains();
+	const Common::String autoSaveName = Common::convertFromU32String(_("Autosave"));
```
Comment (criezy) [inline gui/options.cpp:2500]: Is this variable actually used anywhere?

**Review by criezy — COMMENTED** [review]
The only worry that I have is about performances if the user has added a lot of games in ScummVM and disk access is slow. But I don't know if that can be an issue in practice.

**Comment by orgads** [comment]
> Typo on the commit message: enaling -> enabling

Fixed. Thanks for spotting.

File: gui/options.cpp (hunk 1 above)
Comment (orgads) [reply gui/options.cpp:2500]: Good point. Leftovers from previous revisions.

**Comment by orgads** [comment]
> The only worry that I have is about performances if the user has added a lot of games in ScummVM and disk access is slow. But I don't know if that can be an issue in practice.

Good point. Notice it is only executed when changing from zero to non-zero.

But it might be a good idea to limit the entry count. I suppose 10 entries are enough?

**Comment by sev-** [comment]
I think this is fine. It is more important to not lose a savegame than wait a little bit. Merging.

**Pull request merged**